lib/model/CMetricModel.cc

```diff
@@ -305,12 +307,23 @@ void CMetricModel::sample(core_t::TTime startTime,
                                                    priorWeights[i]);
                 }
 
+                const auto createAndAddAnnotation = [&](core_t::TTime t,
+                                                        const std::string& annotation) {
+                    m_CurrentBucketStats.s_Annotations.emplace_back(
+                        t, annotation, gatherer.searchKey().detectorIndex(),
+                        gatherer.searchKey().partitionFieldName(),
+                        gatherer.partitionFieldValue(),
+                        gatherer.searchKey().overFieldName(), "",
```

Comment:
You could use `EMPTY_STRING` instead of `""`.  It's a static constant in the base class and avoids a call to `strlen` on the `const char*`.